frontend/src/pages/modelServing/customServingRuntimes/CustomServingRuntimeEnabledToggle.tsx

```diff
@@ -0,0 +1,47 @@
+import * as React from 'react';
+import { Switch } from '@patternfly/react-core';
+import { TemplateKind } from '~/k8sTypes';
+import { toggleTemplateEnabledStatus } from '~/api';
+import useNotification from '~/utilities/useNotification';
+import { getTemplateEnabled } from './utils';
+
+type CustomServingRuntimeEnabledToggleProps = {
+  template: TemplateKind;
+};
+
+const CustomServingRuntimeEnabledToggle: React.FC<CustomServingRuntimeEnabledToggleProps> = ({
+  template,
+}) => {
+  const [isEnabled, setEnabled] = React.useState(getTemplateEnabled(template));
+  const [isLoading, setLoading] = React.useState(false);
+  const notification = useNotification();
+
+  const handleChange = (checked: boolean) => {
+    setLoading(true);
+    toggleTemplateEnabledStatus(template.metadata.name, template.metadata.namespace, checked)
+      .then(() => {
+        setEnabled(checked);
+        setLoading(false);
+      })
+      .catch((e) => {
+        notification.error(
+          `Error ${checked ? 'enable' : 'disable'} the serving runtime`,
+          e.message,
+        );
+        setEnabled(!checked);
+        setLoading(false);
```

Comment:
Just out of curiosity, cause I don't see we use the pattern in the code, why don't use finally here to call the `setLoading(false)`, that way you avoid duplicated code.